ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+* Restricted syntax – Has limited query capabilities, compatible with RedisJSON v1.
+
+If a query path starts with '$', the enhanced syntax is used. Otherwise, the restricted syntax is used. A query using 
+the enhanced syntax always returns an array of values, while a restricted-syntax query always returns a single value. 
+
+![](./images/jsonpath_syntax.png)
+
+For details and examples, see https://docs.aws.amazon.com/memorydb/latest/devguide/json-document-overview.html#json-path-syntax.
+
+### ACL
+
+ValkeyJSON introduces a new ACL category - @json. The category has 22 JSON commands. No existing Valkey commands 
+are members of the @json category. 
+
+There are 4 existing ACL categories which are updated to include new JSON commands: @read, @write, @fast, @slow. The 
+table below has a column for each of these categories and a row for each command. If the cell contains a “y” then that 
+command must be added into that category. All of the other command members of those categories remain unchanged.
+
+![](./images/json_command_acl.png)
+
+### Info metrics
+
+Info metrics are visible through the “info json” or “info modules” command.
+
+| Name                    | 	Description                                                           |
+|:------------------------|:---------------------------------------------------------------------|
+| json_total_memory_bytes | 	Total amount of memory allocated to JSON documents and meta data. |
+| json_num_documents	  | Number of JSON keys                                                   | 
+
+### Configs
+
+![](./images/json_configs.png)
+
+### Document Size Limit
+
+JSON documents are stored internally in a format optimized for rapid access and modification. This format typically 
+results in consuming somewhat more memory than the equivalent serialized representation of the same document does. 
+The consumption of memory by a single JSON document is limited to 64MB by default, which is the size of the in-memory 
+data structure, not the serialized JSON string. The value can be overridden by module config json.max-document-size.
+
+The amount of memory consumed by a JSON document can be inspected by 
+using the “JSON.DEBUG MEMORY" or “MEMORY USAGE” command. “JSON.DEBUG MEMORY <key> [<path>]" can also report the size 
+of a JSON sub-tree.
+
+### Nesting Depth Limit
+
+When a JSON object or array has an element that is itself another JSON object or array, that inner object or array is 
+said to “nest” within the outer object or array. To avoid stack overflow, it's good to have a limit on the nesting depth.
+The default path limit is 128 levels. Any attempt to create a document that contains a nesting depth greater than 128 
+will be rejected with an error message. The value can be overridden by module config json.max-path-limit.
+
+### Module API
+
+ValkeyJSON integrates with Valkey core by implementing Valkey module API.
```

Comment:
```suggestion
ValkeyJSON integrates with Valkey core via calls to the Valkey module API.
```

Reply:
Not exactly "via calls to the Valkey module API". Most of the integration is done through callbacks. Will rephrase it.

Reply:
The callbacks are registered via calls to the Valkey module API, yes?

Your new phrasing, "through Valkey module API callbacks or calls to the Valkey core" still troubles me.  The callback code is in the shared library, not the API.  Calls are made to the module API I hope, not the core.  (We do have at least one module that calls the core.  If the JSON module will call the core that is whole other can of worms.)  Perhaps something less specific like "ValkeyJSON shall be implemented via the [Valkey modules API](https://valkey.io/topics/modules-intro/)", or, even simpler, "... shall be a Valkey module, see ...".

Reply:
"ValkeyJSON shall be implemented via the [Valkey modules API](https://valkey.io/topics/modules-intro/)" sounds good to me.